Add missing artifacts

piotr-iohk opened the pull request:
This pr adds missing artifacts such that service can be built:
 - c-reference-signer submodule
 - docker files
 - update paths in build scripts to point to artifacts

piotr-iohk committed:
add c-reference-signer submodule

piotr-iohk committed:
update paths to c-reference-signer in scripts

piotr-iohk committed:
Point c-reference-signer to b5caef0 (same as MinaProtocol/mina)

piotr-iohk committed:
add docker files

piotr-iohk committed:
add release-docker.sh script (temporarily)

smorci commented on dockerfiles/Dockerfile-delegation-backend-toolchain:
I have not used this since I started working with the delegation backend, it might be used in the Buildkite CI. We can leave it in nonetheless but worth asking what is it for.

smorci commented on dockerfiles/Dockerfile-delegation-backend:
This will be removed once we finish the GHA workflow

smorci replied:
FYI, not a blocker

smorci commented on scripts/release-docker.sh:
This file is for every Mina artifact and also is coupled to GCP. I think we should refactor it to make sense in our limited scope of delegation backend. 

To make it, not upload, just build and save the image locally. Than the developer can push it with their credentials to our ECR. 

Once the workflow is ready this will be removed.

piotr-iohk replied:
Makes sense! :+1:

smorci reviewed (changes requested):
I would like to ask you to slim down the `release-docker.sh` script if possible.

piotr-iohk committed:
Simplify building scripts

piotr-iohk committed:
update readme with building and testing info

piotr-iohk commented:
@Smorci I've removed `release-docker.sh` and `Dockerfile-delegation-backend-toolchain` and simplified the scripts for building. Also updated README with information about building and testing.

smorci commented on scripts/build.sh:
```diff
@@ -25,22 +25,9 @@ case "$1" in
-    cd ../../..
-    scripts/release-docker.sh -s delegation-backend${1:6} -v "$VERSION"
+  docker)
+    tag=delegation-backend
```
Here, instead of hardcoding the value of `tag` I would like to be able to pass down the tag with an environment variable just as before you could do it with `$VERSION`.  I suggest we enter the environment variable and check for the existence of it.

smorci commented on README.md:
````diff
@@ -89,3 +89,34 @@ On receiving payload on `/submit`, we perform the following validation:
+[nix-shell]$ make
+```
+
+### Building the Docker Image
````
If you implement the `$TAG` change mentioned in the `build.sh` reflect that here aswell.

piotr-iohk committed:
use TAG when building docker

smorci reviewed (approved):
Perfect! Thank you for you patience!
